src/foundry/client/core/socket.d.mts

```diff
@@ -5,18 +5,13 @@ declare global {
     /**
      * Standardize the way that socket messages are dispatched and their results are handled
      * @param eventName - The socket event name being handled
-     * @param request   - Data provided to the Socket event
+     * @param request   - Request data provided to the Socket event
      * @returns A Promise which resolves to the SocketResponse
      */
-    static dispatch(eventName: string, request: SocketInterface.Request): Promise<SocketInterface.Response>;
-
-    /* -------------------------------------------- */
-
-    /**
-     * Handle an error returned from the database, displaying it on screen and in the console
-     * @param err - The provided Error message
-     */
-    protected static _handleError(err: Error): Error;
+    static dispatch(
+      eventName: string,
+      request: SocketInterface.Request | Record<string, unknown> | number | string | boolean,
```

Comment:
If you look at the use cases, there's two categories
- `ClientBackend.#dispatchRequest`, which passes a `DocumentSocketRequest`
- There's a `manageCompendium` socket that can get called in CompendiumCollection which is something else we can type

Foundry really redid the types for the sockets and we can adjust ours; most of the interfaces at the bottom of this file seem to no longer be needed.

Reply:
So, I traced this through and it calls...

`game.socket.emit(eventName, request, ...)` which I traced back to the socket package and it appears to take anything as the request (though I suspect it can't be a function).  Foundry has it typed here as `DocumentSocketRequest | object` (which was a change from v11 which only had `SocketRequest`, which I now realize isn't the same thing).

So I can redefine `SocketRequest` in `types.d.mts` to be the new `DocumentSocketRequest` typedef, but a) it will still be much broader than SocketInterface.Request is and b) it doesn't resolve whether we should allow other objects.

Are you suggesting we should limit it further (like we had before with just SocketInterface.Request which was very strict?)

Reply:
Yeah, when Foundry writes `object` they mean a Record of some shape. The other two instances

As for the `game.socket.emit`, these database calls are subject to server side validation

Reply:
It just seemed odd that they expanded the type here to object vs. just the `DocumentSocketRequest`.  I'll take a look at it tomorrow

Reply:
The ManageCompendiums event does not conform to the DocumentSocketRequest interface/typedef. We should write our own interface for it.